File: robin/python/level1/study3/budget.py
def solution(d, budget):
    answer = 0
    d.sort()
    for i in d:
        if i <= budget:
            answer += 1
            budget -= i
        else:
            break

    return answer


if __name__ == '__main__':
    d = [1, 3, 2, 5, 4]
    # d = [1,3,4,6,7]
    budget = 9
    print(solution(d, budget))

File: robin/java/src/programmers/level1/study3/Budget.java
package programmers.level1.study3;

import java.util.Arrays;

public class Budget {

    public static int solution(int[] d, int budget) {
        int answer = 0;

        Arrays.sort(d);

        for (int dOne : d) {
            if (dOne <= budget) {
                answer += 1;
                budget -= dOne;
            } else {
                break;
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        int[] d = {1, 3, 2, 5, 4};
        int budget = 9;
        System.out.println(solution(d, budget));
    }
}
